elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/SQLQueryConstructor.java

```diff
@@ -320,22 +324,25 @@ private String extractOrderBy(Map<Path, Sorting.SortOrder> sortClauses, SQLQuery
 
         return " ORDER BY " + sortClauses.entrySet().stream()
                 .map((entry) -> {
-                    Path expandedPath = expandJoinToPath(entry.getKey());
+                    Path path = entry.getKey();
                     Sorting.SortOrder order = entry.getValue();
 
-                    Path.PathElement last = expandedPath.lastElement().get();
+                    Path.PathElement last = path.lastElement().get();
 
                     MetricFunctionInvocation metric = template.getMetrics().stream()
                             // TODO: filter predicate should support alias
                             .filter(invocation -> invocation.getAlias().equals(last.getFieldName()))
                             .findFirst()
                             .orElse(null);
 
-                    String orderByClause = metric == null
-                            ? generateColumnReference(expandedPath, dictionary)
-                            : metric.getFunctionExpression();
+                    // if the sorted field is a metric field on this queried table, it should be present in the request
```

Comment:
What is metric != null but the last type doesn't match the queried table class?

Reply:
The metric is searched using alias. There can be a dimension class field that have the same as the projected metric.